### ISSUE TITLE
DB change in node.js 

### ISSUE THREAD

**Issue by uppalapati41** [nodeschool/discussions#940]
Hello I have a project where am using SQL server driver(tedious) but am not able to understand how to get db changed values. Tried using dbmon but was not able to get result set. Can any one help me out.

**Comment by martinheidegger** [comment]
Hello @uppalapati41 
"get db changed values"? It would help if you could write about what you want to do. This is what I understand from your request: "You would like to add a listener to a database change event and you havn't found out a way to do that. You tried to use tedious but it seems to have no way to track the changes and you tried to use dbmon where you did get events but it didn't contain any data."

There are several problems with your question to me:
1. "SQL Server": There are many SQL Servers out there, which one are you using?
2. "db changed values": To my knowledge here is no standard SQL abstraction to listen to change events
3. Tedious and dbmon do not do the same thing

**Comment by uppalapati41** [comment]
I have a project where I will be showing details about ship geo-location for that we went with node and our db is SQL server 2012. Drivers which support SQL server 2012 and above is tedious so used it. I was able to get data in order to listen to changes in db with out long polling I though I can use dbmon which supports different databases in order to listen any crud operation's in db. But was not able to figure it out.

What I finally need is when ever there is any crud operation occurs in db I should get a event back so that I can make call. Is there any other way other then long polling.

**Comment by martinheidegger** [comment]
@uppalapati41 This is a very specific question to an interesting topic. However it seems less related to the NodeSchool or SQL abstraction modules than it is to the mssql database driver abstraction. Perhaps you better post your question [here](https://github.com/patriksimek/node-mssql)? (Disclaimer: I have no idea how/if change events work with mssql at the first place...)

**Comment by uppalapati41** [comment]
Hey thanx for info. Can you even clear one more doubt!!
How many users does node.js & socket.io together can support. I heard it is some where only 100.
What if i use rooms & namespace concept in socket.io? Will it let me handle more users!! [I know it is even depend on server speed]

**Comment by Sequoia** [comment]
Hey @uppalapati41! As @martinheidegger mentioned this is an interesting question but not nodeschool related-- try StackOverflow or the repo linked. Good luck! :smile_cat:

**Comment by martinheidegger** [comment]
@uppalapati41 we have had a Project with 100000 Users using socket.io but you Need to apply regular techniques for scaling. In the end throughput, memory and CPUs become real bottlenecks.

**Comment by uppalapati41** [comment]
Have came across package called autobahn it allows 200+k users. Do u have
any idea about that??

**Comment by martinheidegger** [comment]
I read about it a while ago. 200k+ are "no problem" with nodejs either. Usually the scaling issues rarely happen in the web socket library itself but in other details (like db access)

**Comment by uppalapati41** [comment]
Hey thanx martin for taking your time out. So one last thing I will be
exposed with web service to get data so can I go with socket.io or
socket.io-rpc that going to manageable rite?? Thanx!!

**Comment by martinheidegger** [comment]
Can you rephrase the question? I don't understand what you mean

**Comment by uppalapati41** [comment]
We will be given with web service so that can I use socket.io? Am confused
between which package to use autobahn or socket.io completed initial
analysis on both. Need expert advice. This is first time we are trying
project in node.js

**Comment by martinheidegger** [comment]
I didn't research autobahn in detail but socketio works for many things. I guess both have their caveats.nodejs has a learning curve but it's not as steep as with other programming environments.

**Comment by uppalapati41** [comment]
Cool thanx for info!!

**Comment by martinheidegger** [comment]
No problem.

**Comment by muhammetvrl** [comment]
Did you solve the problem? I'm having the same problem. link in the module need mssql version.can you help?  https://www.npmjs.com/package/mysql-events
